### DISCUSSION TITLE
Reuse InfluxDB Flux query between panels

### DISCUSSION THREAD

**Discussion by Joe-Influx** [grafana/grafana#49546] [Dashboards, Panels & Visualizations]
I am trying to reuse the same query in two Grafana panels:
One panel needs “all the data”, while the other panel needs only “a subset of the data” (the subset that could be easily identified by the “name” of one of the yield() in the query).  I’m wondering if there is a way to let the second query only see the data from that yield().

The context is:
When running a query with InfluxDB and using “yield(name: <some_string>)“, the resulting CSV includes a “result” column that maps to that <some_string>.

When creating a dashboard panel in Grafana, it is possible to specify a Flux query (which might include some number of “yield()“) in the “Query” tab.

I need a functionality equivalent to the “Filter data by query” in the “Transform” tab of Grafana, to filter data based on the <some_string> of yield().

I was wondering if there can be a plugin or a workaround to use a Grafana Transform to select only the data returned by a yield() in the context of a larger query that might have returned more data.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82996. Feel free to continue the discussion around this there. Thank you!
